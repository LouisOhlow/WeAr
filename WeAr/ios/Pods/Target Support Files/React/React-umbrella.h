#ifdef __OBJC__
#import <UIKit/UIKit.h>
#else
#ifndef FOUNDATION_EXPORT
#if defined(__cplusplus)
#define FOUNDATION_EXPORT extern "C"
#else
#define FOUNDATION_EXPORT extern
#endif
#endif
#endif

#import "React/Base/RCTAssert.h"
#import "React/Base/RCTBridge.h"
#import "React/Base/RCTBridgeDelegate.h"
#import "React/Base/RCTBridgeModule.h"
#import "React/Base/RCTCache.h"
#import "React/Base/RCTConvert.h"
#import "React/Base/RCTDefines.h"
#import "React/Base/RCTEventDispatcher.h"
#import "React/Base/RCTFPSGraph.h"
#import "React/Base/RCTFrameUpdate.h"
#import "React/Base/RCTInvalidating.h"
#import "React/Base/RCTJavaScriptExecutor.h"
#import "React/Base/RCTJavaScriptLoader.h"
#import "React/Base/RCTJSMethodRegistrar.h"
#import "React/Base/RCTKeyboardObserver.h"
#import "React/Base/RCTKeyCommands.h"
#import "React/Base/RCTLog.h"
#import "React/Base/RCTModuleData.h"
#import "React/Base/RCTModuleMap.h"
#import "React/Base/RCTModuleMethod.h"
#import "React/Base/RCTPerformanceLogger.h"
#import "React/Base/RCTPerfStats.h"
#import "React/Base/RCTProfile.h"
#import "React/Base/RCTRootView.h"
#import "React/Base/RCTSparseArray.h"
#import "React/Base/RCTTouchHandler.h"
#import "React/Base/RCTURLRequestDelegate.h"
#import "React/Base/RCTURLRequestHandler.h"
#import "React/Base/RCTUtils.h"
#import "React/Executors/RCTContextExecutor.h"
#import "React/Executors/RCTWebViewExecutor.h"
#import "React/Layout/Layout.h"
#import "React/Modules/RCTAccessibilityManager.h"
#import "React/Modules/RCTAlertManager.h"
#import "React/Modules/RCTAppState.h"
#import "React/Modules/RCTAsyncLocalStorage.h"
#import "React/Modules/RCTDevLoadingView.h"
#import "React/Modules/RCTDevMenu.h"
#import "React/Modules/RCTExceptionsManager.h"
#import "React/Modules/RCTPointAnnotation.h"
#import "React/Modules/RCTRedBox.h"
#import "React/Modules/RCTSourceCode.h"
#import "React/Modules/RCTStatusBarManager.h"
#import "React/Modules/RCTTiming.h"
#import "React/Modules/RCTUIManager.h"
#import "React/Views/RCTActivityIndicatorViewManager.h"
#import "React/Views/RCTAnimationType.h"
#import "React/Views/RCTAutoInsetsProtocol.h"
#import "React/Views/RCTBorderDrawing.h"
#import "React/Views/RCTComponent.h"
#import "React/Views/RCTComponentData.h"
#import "React/Views/RCTConvert+CoreLocation.h"
#import "React/Views/RCTConvert+MapKit.h"
#import "React/Views/RCTDatePickerManager.h"
#import "React/Views/RCTMap.h"
#import "React/Views/RCTMapManager.h"
#import "React/Views/RCTModalHostView.h"
#import "React/Views/RCTModalHostViewController.h"
#import "React/Views/RCTModalHostViewManager.h"
#import "React/Views/RCTNavigator.h"
#import "React/Views/RCTNavigatorManager.h"
#import "React/Views/RCTNavItem.h"
#import "React/Views/RCTNavItemManager.h"
#import "React/Views/RCTPicker.h"
#import "React/Views/RCTPickerManager.h"
#import "React/Views/RCTPointerEvents.h"
#import "React/Views/RCTProgressViewManager.h"
#import "React/Views/RCTScrollableProtocol.h"
#import "React/Views/RCTScrollView.h"
#import "React/Views/RCTScrollViewManager.h"
#import "React/Views/RCTSegmentedControl.h"
#import "React/Views/RCTSegmentedControlManager.h"
#import "React/Views/RCTShadowView.h"
#import "React/Views/RCTSlider.h"
#import "React/Views/RCTSliderManager.h"
#import "React/Views/RCTSwitch.h"
#import "React/Views/RCTSwitchManager.h"
#import "React/Views/RCTTabBar.h"
#import "React/Views/RCTTabBarItem.h"
#import "React/Views/RCTTabBarItemManager.h"
#import "React/Views/RCTTabBarManager.h"
#import "React/Views/RCTTextDecorationLineType.h"
#import "React/Views/RCTView.h"
#import "React/Views/RCTViewControllerProtocol.h"
#import "React/Views/RCTViewManager.h"
#import "React/Views/RCTWebView.h"
#import "React/Views/RCTWebViewManager.h"
#import "React/Views/RCTWrapperViewController.h"
#import "React/Views/UIView+Private.h"
#import "React/Views/UIView+React.h"

FOUNDATION_EXPORT double ReactVersionNumber;
FOUNDATION_EXPORT const unsigned char ReactVersionString[];

