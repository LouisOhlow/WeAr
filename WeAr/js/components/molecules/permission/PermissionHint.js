import React from 'react';
import { StyleSheet, View } from 'react-native';
import PermissionButton from '../../atoms/permission/PermissionButton';
import Headline1 from '../../atoms/basics/Headline1';

export default function PermissionHint(props) {
  const { permissionStatus, checkPerm } = props;

  return (
    <View style={styles.permissionView}>
      <View style={styles.permissionHint}>
        <Headline1 text="You need to accept all necessary permissions before you can use the app." />
        <PermissionButton
          permissionStatus={permissionStatus}
          checkPerm={checkPerm}
        />
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  permissionView: {
    width: '100%',
    height: '100%',
    backgroundColor: 'black',
    justifyContent: 'center',
  },
  permissionHint: {
    height: 159,
    width: 259,
    justifyContent: 'space-between',
    alignSelf: 'center'
  }
});
