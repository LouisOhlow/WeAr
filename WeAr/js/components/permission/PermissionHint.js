import React from 'react';
import { StyleSheet, View } from 'react-native';
import PermissionButton from './PermissionButton';
import PermissionText from './PermissionText';

export default function PermissionHint(props) {
  const { permissionStatus, checkPerm } = props;

  return (
    <View style={styles.permissionView}>
      <View style={styles.permissionHint}>
        <PermissionText />
        <PermissionButton
          permissionStatus={permissionStatus}
          checkPerm={checkPerm}
        />
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  permissionView: {
    width: '100%',
    height: '100%',
    backgroundColor: 'black',
    justifyContent: 'center',
  },
  permissionHint: {
    height: 159,
    width: 259,
    justifyContent: 'space-between',
    alignSelf: 'center',
  },
});
