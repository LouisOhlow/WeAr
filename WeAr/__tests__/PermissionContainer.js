import renderer from 'react-test-renderer';
import React from 'react';
import PermissionHint from '../js/components/molecules/permission/PermissionHint';

jest.mock('react-native-permissions', () => ({
  PERMISSIONS: jest.fn()
}));

jest.mock('../js/utils/permission/Permission', () => ({
  Permission: { checkPermissionStatus: async () => 'granted' },
  PermissionHandler: jest.fn(),
  PERMISSION_TYPE: jest.requireActual()
}));

it('test PermissionHint Snapshot with denied permission', () => {
  const tree = renderer
    .create(<PermissionHint permissionStatus="denied" checkPerm={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('test PermissionHint Snapshot with blocked permission', () => {
  const tree = renderer
    .create(<PermissionHint permissionStatus="blocked" checkPerm={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
