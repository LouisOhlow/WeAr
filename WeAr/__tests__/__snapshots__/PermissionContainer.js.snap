// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test PermissionHint Snapshot with blocked permission 1`] = `
<View
  style={
    Object {
      "backgroundColor": "black",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "height": 159,
        "justifyContent": "space-between",
        "width": 259,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        You need to accept all necessary permissions before you can use the app.
      </Text>
    </View>
    <View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "grey",
            "borderRadius": 10,
            "elevation": 8,
            "opacity": 1,
            "paddingHorizontal": 12,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 20,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          GO TO SETTINGS!
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`test PermissionHint Snapshot with denied permission 1`] = `
<View
  style={
    Object {
      "backgroundColor": "black",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "height": 159,
        "justifyContent": "space-between",
        "width": 259,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        You need to accept all necessary permissions before you can use the app.
      </Text>
    </View>
    <View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "grey",
            "borderRadius": 10,
            "elevation": 8,
            "opacity": 1,
            "paddingHorizontal": 12,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 20,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          GOT IT!
        </Text>
      </View>
    </View>
  </View>
</View>
`;
